src/database.cpp

```diff
@@ -50,81 +81,44 @@ bool Database::connect()
 
 bool Database::beginTransaction()
 {
-	if (!executeQuery("BEGIN")) {
-		return false;
-	}
-
 	databaseLock.lock();
```

Comment:
This could lead to a race condition. The `lock` must happen before calling `Database::executeQuery`.